dashboard/src/components/AppList/CustomResourceListItem.tsx

```diff
@@ -6,19 +6,20 @@ import InfoCard from "../InfoCard";
 
 interface ICustomResourceListItemProps {
   resource: IResource;
-  csv?: IClusterServiceVersion;
+  csv: IClusterServiceVersion;
 }
 
 class CustomResourceListItem extends React.Component<ICustomResourceListItemProps> {
   public render() {
     const { resource, csv } = this.props;
-    const icon = csv?.spec.icon
+    const icon = csv.spec.icon
       ? `data:${csv.spec.icon[0].mediatype};base64,${csv.spec.icon[0].base64data}`
       : placeholder;
+    const crd = csv.spec.customresourcedefinitions.owned.find(c => c.kind === resource.kind);
     return (
       <InfoCard
         key={resource.metadata.name}
-        link={`/operators-instances/ns/${resource.metadata.namespace}/${resource.metadata.name}`}
+        link={`/operators-instances/ns/${resource.metadata.namespace}/${csv.metadata.name}/${crd?.name}/${resource.metadata.name}`}
```

Comment:
Won't the `crd?.name` leave either the string 'undefined' or at best, '' so there will be two slashes?

Reply:
yes, the thing is that `crd` will never be empty if `resource` is populated since the crd is required to obtain the resource. Typescript is not helping here a lot but I will try to make it more readable.